plugins/content_distribution/providers/facebook/lib/FacebookDistributionEngine.php

```diff
@@ -0,0 +1,150 @@
+<?php
+
+/**
+ * @package plugins.facebookDistribution
+ * @subpackage lib
+ */
+class FacebookDistributionEngine extends DistributionEngine implements IDistributionEngineSubmit, IDistributionEngineCloseSubmit, IDistributionEngineUpdate
+{
+	protected $appId;
+	protected $appSecret;
+	
+	/* (non-PHPdoc)
+	 * @see DistributionEngine::configure()
+	 */
+	public function configure()
+	{
+		parent::configure();
+		if (isset(KBatchBase::$taskConfig->params->facebook))
+		{
+			if (isset(KBatchBase::$taskConfig->params->facebook->appId))
```

Comment:
I suggest that you'll add the app id and secret to the job provider data and load it from kConf instead of having two places that should be configured.